### DISCUSSION TITLE
timezones

### DISCUSSION THREAD

**Discussion by drtonyr** [wasp-os/wasp-os#456] [Q&A]
I've grepped the WASP-OS source code and read https://github.com/orgs/micropython/discussions/12378 to get an idea of timezones for WASP-OS.   As far as I can tell, there is no support and there's not going to be any.   That is, WASP-OS uses localtime and when the clocks change we have to change the time manually, call `tools/wasptool --rtc` or use gadgetbridge.

This is what we have to play with:
```
think0 tonyr: tools/wasptool --console
Connected to P8 (F2:61:81:68:9B:F2).
Exit console using Ctrl-X.
>>> import time
>>> dir(time)
['__class__', '__name__', 'localtime', 'mktime', 'sleep', 'sleep_ms', 'sleep_us', 'ticks_add', 'ticks_diff', 'ticks_ms', 'ticks_us']
>>> import watch
>>> dir(watch.rtc)
['__class__', '__init__', '__module__', '__qualname__', 'update', '__dict__', 'counter', 'time', 'get_localtime', 'set_localtime', 'get_uptime_ms', 'get_time', 'uptime', 'lastcount', 'offset', '_uptime']
>>> watch.rtc.time()
748613509
```
I've done a load of maths and modelling for tide times and I'd like to get it onto my watch and make it available.  For this to work, I need something like epoch time (UTC), not the displayed localtime.

For epoch time, I note that watch.rtc.time() uses 1 Jan 2000 not 1 Jan 1970, so it would probably be a good idea not to code in either date but to call time.mktime() for a known date and use that as an offset.

For timezones, I guess I'm on my own.   I can code in the UK timezones (in a lightweight fashion) and if anyone likes my app then we can fix up for other locations later.

Has anyone else hacked around timezones?

Tony

**Comment by thiswillbeyourgithub** [comment]
The only relevant thing I have to say is that when coding sleep_tk I wanted to use a unix time timestamp to date the nights and had to use this offset value to make both rtc.time() and my computer's time.time() agree IIRC.

```
_TIMESTAMP = const(946684800)  # unix time and time used by wasp os don't have the same reference date
```

**Reply by drtonyr** [reply]
Ta.  It's a bit of a pain isn't it.   And I thought this would be the easy bit!
